lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -550,6 +554,75 @@ def _graded_scorable_blocks_to_header(cls, course):
 
 
 class ProblemResponses(object):
+
+    @classmethod
+    def _build_problem_list(cls, course_blocks, root, path=None):
+        """
+        Generate a tuple of display names, block location paths and block keys
+        for all problem blocks under the ``root`` block.
+
+        Arguments:
+            course_blocks (BlockStructureBlockData): Block structure for a course.
+            root (UsageKey): This block and its children will be used to generate
+                the problem list
+            path (List[str]): The list of display names for the parent of root block
+
+        Yields:
+            Tuple[str, List[str], UsageKey]: tuple of a block's display name, path, and
+                usage key
+        """
+        display_name = course_blocks.get_xblock_field(root, 'display_name')
+        if path is None:
+            path = [display_name]
+
+        yield display_name, path, root
+
+        for block in course_blocks.get_children(root):
+            display_name = course_blocks.get_xblock_field(block, 'display_name')
+            for result in cls._build_problem_list(course_blocks, block, path + [display_name]):
+                yield result
+
+    @classmethod
+    def _build_student_data(cls, user_id, course_id, problem_location):
+        """
+        Generate a list of problem responses for all problem under the
+        ``problem_location`` root.
+
+        Arguments:
+            user_id (int): The user id for the user generating the report
+            course_id (CourseKey): The ``CourseKey`` for the course whose report
+                is being generated
+            problem_location (str): The generated report will include this
+                block and it child blocks.
+
+        Returns:
+              List[Dict]: Returns a list of dictionaries containing the student
+                data which will be included in the final csv.
+        """
+        problem_key = UsageKey.from_string(problem_location).map_into_course(course_id)
+        user = get_user_model().objects.get(pk=user_id)
+        course_blocks = get_course_blocks(user, problem_key)
+
+        student_data = []
+        max_count = settings.FEATURES.get('MAX_PROBLEM_RESPONSES_COUNT')
+        for title, path, block_key in cls._build_problem_list(course_blocks, problem_key):
+            # Sequential blocks are filtered out since they include position state
+            # which isn't useful in this report.
+            if block_key.block_type == 'sequential':
```

Comment:
Please also exclude `chapter` type blocks, for the same reason. In the edX demo course, you can test this out with the key `block-v1:edX+DemoX+Demo_Course+type@chapter+block@interactive_demonstrations`

Reply:
OK. Updated the PR.